chamber/utils/transaction.py

```diff
@@ -58,7 +64,10 @@ def register(self, callable):
             self._callable_list.append(callable)
 
     def handle_all(self):
-        for callable in self._callable_list:
+        while self._callable_list:
+            callable = self._callable_list.pop(0)
```

Comment:
Why are you removing it?

Reply:
Because I need to remove functions which will be called